### DISCUSSION TITLE
How to do matrix multiplication in c++

### DISCUSSION THREAD

**Discussion by charleskhazoom** [casadi/casadi#3315] [Q&A] [answered]
I realized that the * operator in c++ is for element-size multiplication. How to do matrix multiplication ? I tried :
 
 
  `casadi::MX x = casadi::MX::sym("x", 10, 10);
    casadi::MX y = casadi::MX::sym("y", 10, 10);
    casadi::MX z = casadi::mtimes(x,y);`
    

    error: ‘mtimes’ is not a member of ‘casadi’
     casadi::MX z = casadi::mtimes(x,y);

**Comment by jgillis** [comment, marked as answer]
Drop the namespace on mtimes. It's a friend function dispatches through argument dependant lookup: https://www.appsloveworld.com/cplus/100/220/why-is-a-friend-function-not-treated-as-a-member-of-a-namespace-of-a-class-it-was?utm_content=cmp-true
